File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]):
        res = -1
        min_dist = 99999999
        for n, i in enumerate(points):
            if i[0] == x or i[1] == y:
                _ = abs(x-i[0]) + abs(y-i[1])
                if _ < min_dist:
                    min_dist = _
                    res = n
        return res

File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.go
func nearestValidPoint(x int, y int, points [][]int) int {
    res := -1
    min_dist := float64(99999)
    var tmp float64
    for i:=0;i<len(points);i++{
        if points[i][0] == x || points[i][1] == y{
            tmp = math.Abs(float64(x-points[i][0])) + math.Abs(float64(y-points[i][1]))
            if tmp < min_dist{
                res = i
                min_dist = tmp
            }
        } 
    }
    return res
}